File: react-frontend/src/types/forecast.ts
export interface DailyForecast {
    date: string;
    temperatureMax: number;
    temperatureMin: number;
    weatherCode: number;
    precipitation: number;
}

export interface ForecastData {
    latitude: number;
    longitude: number;
    daily: {
        time: string[];
        temperature_2m_max: number[];
        temperature_2m_min: number[];
        weather_code: number[];
        precipitation_sum: number[];
    };
}

File: go-backend/models/forecast.go
package models

type ForecastResponse struct {
	Latitude  float64 `json:"latitude"`
	Longitude float64 `json:"longitude"`
	Daily     struct {
		Time             []string  `json:"time"`
		TemperatureMax   []float64 `json:"temperature_2m_max"`
		TemperatureMin   []float64 `json:"temperature_2m_min"`
		WeatherCode      []int     `json:"weather_code"`
		PrecipitationSum []float64 `json:"precipitation_sum"`
	} `json:"daily"`
}
